order/manager.go

```diff
@@ -334,6 +384,59 @@ func (m *Manager) OurNodePubkey() ([33]byte, error) {
 	return m.ourNodeInfo.IdentityPubkey, nil
 }
 
+// validateAndSignTicketForOrder makes sure that the given sidecar ticket is in
+// the correct state for being used in a bid order and has all the necessary
+// information set. We also check that our node initially offered to lease this
+// channel by checking the embedded signature. If everything checks out, we add
+// our signature over the order part to the ticket.
+func (m *Manager) validateAndSignTicketForOrder(ctx context.Context,
+	t *sidecar.Ticket, bid *Bid) error {
+
+	if t.State != sidecar.StateRegistered {
+		return fmt.Errorf("invalid sidecar ticket state: %d", t.State)
+	}
+
+	// In theory a ticket should never be in the "registered" state if the
+	// information in the following checks is missing. But we never know...
+	r := t.Recipient
+	if r == nil || r.NodePubKey == nil || r.MultiSigPubKey == nil {
+		return fmt.Errorf("invalid sidecar ticket, missing recipient " +
+			"information")
+	}
+
+	// Make sure the offer is valid and actually came from us.
+	o := t.Offer
+	if err := sidecar.VerifyOffer(ctx, t, m.cfg.Signer); err != nil {
+		return fmt.Errorf("error verifying sidecar offer: %v", err)
+	}
+
+	ourNodeKeyRaw, err := m.OurNodePubkey()
```

Comment:
I can see how this is useful on the ask side (similar to chan ann's we have two sigs for the on-chain output as well as the node itself), but what does this add from the PoV of the bidder here? Shouldn't this be done with the public key of the account submitting the bid? However in that scenario the order digest is already being signed which _includes_ the side car ticket itself.

Reply:
I guess it locks things down a bit, since it's possible for the node creating the offer to be different than the node creating the bid itself.